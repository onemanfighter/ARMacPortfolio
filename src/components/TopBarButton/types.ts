export type TopBarButtonProps = {
  onClick: () => void;
  icon?: React.ReactElement;
  text: string;
  ariaLabel: string;
};
