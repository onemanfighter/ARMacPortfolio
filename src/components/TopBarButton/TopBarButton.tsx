import { Button, MenuButton, Text } from '@chakra-ui/react';
import { TopBarButtonProps } from './types';

const TopBarButton = ({
  text,
  icon,
  onClick,
  ariaLabel,
}: TopBarButtonProps) => {
  return (
    <MenuButton
      fontSize={'12'}
      px={2}
      py={0.5}
      color={'white'}
      borderRadius={4}
      fontWeight={'600'}
      _active={{
        bg: '#ffffff4f',
      }}
      onClick={onClick}
      aria-label={ariaLabel}
    >
      {text ? <Text>{text}</Text> : null}
      {icon}
    </MenuButton>
  );
};

export default TopBarButton;
