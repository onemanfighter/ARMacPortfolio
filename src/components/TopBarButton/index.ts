export { default as TopBarButton } from './TopBarButton';
