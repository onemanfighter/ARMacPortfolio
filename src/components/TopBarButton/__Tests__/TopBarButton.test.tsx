import { fireEvent, render, screen } from '@testing-library/react';
import TopBarButton from '../TopBarButton';
import { Menu } from '@chakra-ui/react';
import { SearchIcon } from '@assets';

const clickHandler = jest.fn();

describe('TopBar button', () => {
  beforeEach(() => {
    clickHandler.mockReset();
  });

  it('should render for default values', () => {
    const { container } = render(
      <Menu>
        <TopBarButton
          text="abc"
          onClick={clickHandler}
          ariaLabel="abc-component"
        />
      </Menu>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should call click handler when clicked', () => {
    render(
      <Menu>
        <TopBarButton
          text="abc"
          onClick={clickHandler}
          ariaLabel="abc-component"
        />
      </Menu>,
    );

    fireEvent.click(screen.getByRole('button'));

    expect(clickHandler).toHaveBeenCalled();
  });

  it('should render with search icon', () => {
    const { container } = render(
      <Menu>
        <TopBarButton
          text="abc"
          icon={<SearchIcon />}
          onClick={clickHandler}
          ariaLabel="abc-component"
        />
      </Menu>,
    );

    expect(container).toMatchSnapshot();
  });
});
