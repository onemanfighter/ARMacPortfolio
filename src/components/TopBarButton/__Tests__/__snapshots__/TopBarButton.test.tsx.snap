// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar button should render for default values 1`] = `
<div>
  <button
    aria-controls="menu-list-:r1:"
    aria-expanded="false"
    aria-haspopup="menu"
    aria-label="abc-component"
    class="chakra-menu__menu-button css-10l0d22"
    id="menu-button-:r1:"
  >
    <span
      class="css-xl71ch"
    >
      <p
        class="chakra-text css-0"
      >
        abc
      </p>
    </span>
  </button>
</div>
`;

exports[`TopBar button should render with search icon 1`] = `
<div>
  <button
    aria-controls="menu-list-:r5:"
    aria-expanded="false"
    aria-haspopup="menu"
    aria-label="abc-component"
    class="chakra-menu__menu-button css-10l0d22"
    id="menu-button-:r5:"
  >
    <span
      class="css-xl71ch"
    >
      <p
        class="chakra-text css-0"
      >
        abc
      </p>
      <svg
        fill="currentColor"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 18a7.952 7.952 0 004.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0018 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
        />
      </svg>
    </span>
  </button>
</div>
`;
